app/shop.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import annotations
+from typing import Any
+
+from app.point import Point
+
+
+class Shop:
+    def __init__(
+            self,
+            name: str,
+            location: Point,
+            products: dict[str, float | int]
+    ) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+
+    @classmethod
+    def create_shop(
+            cls,
+            shop_info: dict[str, str | list[int] | dict[str, (int | float)]]
+    ) -> Shop:
+        return cls(
+            name=shop_info["name"],
+            location=Point.create_from_location(shop_info["location"]),
+            products=shop_info["products"]
+        )
+
+    def sell(self, customer: Any) -> None:
+        print("Date: 04/01/2021 12:33:41")
+        print(f"Thanks, {customer.name}, for your purchase!")
+        print("You have bought:")
+        for name, amount in customer.product_cart.items():
+            product_price = amount * self.products[name]
```

Comment:
To prevent `KeyError` exceptions when accessing `self.products[name]`, consider checking if `name` exists in `self.products` before accessing it. You can use `self.products.get(name, 0)` to provide a default value if the key is missing. This will make the code more robust against missing product names.